source/daq/egg3_reader.hh

```diff
@@ -0,0 +1,77 @@
+/*
+ * egg3_reader.hh
+ *
+ *  Created on: Dec 14, 2017
+ *      Author: laroque
+ */
+
+#ifndef PSYLLID_EGG3_READER_HH_
+#define PSYLLID_EGG3_READER_HH_
+
+#include "memory_block.hh"
+#include "node_builder.hh"
+#include "time_data.hh"
+
+#include "producer.hh"
+#include "M3Monarch.hh"
```

Comment:
Instead of including M3Monarch.hh and time_data.hh here in the header, forward declare the relevant classes and then include the files as necessary in the source file.  You only refer to pointers to the classes here, so you don't need to include the actual files yet.